### PR TITLE
[ProgressBar-203-376] Add all content data and logic

### PR DESCRIPTION
**Pull request by robergro**
This PR contains all needed data for the three versions of the ProgressBar (default, double and indeterminate)
- AccessibilityIdentifier
- Constants
- Enums
- Model
- UseCase
- ViewModel

### CODE REVIEW

File: core/Sources/Components/ProgressBar/ViewModel/Double/ProgressDoubleBarViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            intent: intent,
+            shape: shape,
+            getColorsUseCase: ProgressDoubleBarGetColorsUseCase()
+        )
+    }
+}
+
+// MARK: - Extension
+
+extension ProgressDoubleBarViewModel: ProgressBarValueViewModel {
```
Comment (jacklyn-situmorang) [inline core/Sources/Components/ProgressBar/ViewModel/Double/ProgressDoubleBarViewModel.swift:31]: Do we need this empty extension?

File: core/Sources/Components/ProgressBar/ViewModel/Single/ProgressBarViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            intent: intent,
+            shape: shape,
+            getColorsUseCase: ProgressBarGetColorsUseCase()
+        )
+    }
+}
+
+// MARK: - Extension
+
+extension ProgressBarViewModel: ProgressBarValueViewModel {
```
Comment (jacklyn-situmorang) [inline core/Sources/Components/ProgressBar/ViewModel/Single/ProgressBarViewModel.swift:31]: Also here?

**Review by jacklyn-situmorang — COMMENTED** [review]
In general everything looks great! I would just suggest renaming `ProgressIndeterminateBar....` to `ProgressBarIndeterminate...`

File: core/Sources/Components/ProgressBar/ViewModel/Single/ProgressBarViewModel.swift (hunk 2 above)
Comment (robergro) [reply core/Sources/Components/ProgressBar/ViewModel/Single/ProgressBarViewModel.swift:31]: Same answer :)

File: core/Sources/Components/ProgressBar/Enum/Indeterminate/ProgressIndeterminateBarAnimationType.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
+//
+//  ProgressIndeterminateBarAnimationType.swift
+//  SparkCore
+//
+//  Created by robin.lemaire on 28/09/2023.
+//  Copyright © 2023 Adevinta. All rights reserved.
+//
+
+enum ProgressIndeterminateBarAnimationType: Equatable {
```
[the remark this answers is not in the document]
Comment (robergro) [reply core/Sources/Components/ProgressBar/Enum/Indeterminate/ProgressIndeterminateBarAnimationType.swift]: I replace all ProgressIndeterminateBar to ProgressBarIndeterminate :)

File: core/Sources/Components/ProgressBar/Enum/Indeterminate/ProgressIndeterminateBarStatus.swift [hunk 4]
```diff
@@ -0,0 +1,12 @@
+//
+//  ProgressIndeterminateBarStatus.swift
+//  SparkCore
+//
+//  Created by robin.lemaire on 29/09/2023.
+//  Copyright © 2023 Adevinta. All rights reserved.
+//
+
+enum ProgressIndeterminateBarStatus: Equatable {
```
[the remark this answers is not in the document]
Comment (robergro) [reply core/Sources/Components/ProgressBar/Enum/Indeterminate/ProgressIndeterminateBarStatus.swift]: Same here

File: core/Sources/Components/ProgressBar/Enum/Indeterminate/ProgressIndeterminateBarAnimationType.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (robergro) [reply core/Sources/Components/ProgressBar/Enum/Indeterminate/ProgressIndeterminateBarAnimationType.swift]: And same for the ProgressDoubleBar (now it's ProgressBarDouble)

**Review by jacklyn-situmorang — APPROVED** [review]

**Pull request merged**